### ISSUE TITLE
Method of regexp matching detached is not working with newer versions of git

### ISSUE THREAD

**Issue by ZenithDK** [randy3k/Git-StatusBar#1]
```
DPI scale: 1
startup, version: 3065 linux x64 channel: stable
executable: /opt/sublime_text/sublime_text
working dir: /
packages path: /home/lbigum/.config/sublime-text-3/Packages
state path: /home/lbigum/.config/sublime-text-3/Local
hardware concurrency: 8
zip path: /opt/sublime_text/Packages
zip path: /home/lbigum/.config/sublime-text-3/Installed Packages
found 7 files for base name Default.sublime-keymap
found 1 files for base name Default.sublime-mousemap
found 6 files for base name Main.sublime-menu
loading bindings
loading pointer bindings
found 1 files for base name Default.sublime-theme
theme loaded
app ready
wrote startup cache, added files: 3 orphaned files: 0 total files: 187 cache hits: 184
pre session restore time: 0.115332
using gamma: 1 (err: 0)
startup time: 0.216658
first paint time: 0.228159
launching: /opt/sublime_text/plugin_host
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_file_settings
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-ssl-linux
Linux SSL: successfully loaded _ssl module for libssl.so.1.0.0
reloading plugin 0_package_control_loader.02-bz2
reloading plugin Git-StatusBar.git-statusbar
reloading plugin Git.add
reloading plugin Git.annotate
reloading plugin Git.commit
reloading plugin Git.diff
reloading plugin Git.flow
reloading plugin Git.git
reloading plugin Git.history
reloading plugin Git.repo
reloading plugin Git.stash
reloading plugin Git.status
reloading plugin Git.statusbar
reloading plugin GitGutter.git_gutter
reloading plugin GitGutter.git_gutter_change
reloading plugin GitGutter.git_gutter_compare
reloading plugin GitGutter.git_gutter_events
reloading plugin GitGutter.git_gutter_handler
reloading plugin GitGutter.git_helper
reloading plugin GitGutter.view_collection
reloading plugin Package Control.Package Control
loaded 837 snippets
reloading plugin Package Control.bootstrap
reloading plugin StyleToken.StyleToken
reloading plugin Anaconda.anaconda
reloading plugin Anaconda.version
reloading plugin Toks.SublimeToks
plugins loaded
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 303, in on_activated
    callback.on_activated(v)
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 114, in on_activated
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 101, in update_status_bar
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 82, in badge
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 59, in branch
AttributeError: 'NoneType' object has no attribute 'group'
Package Control: Skipping automatic upgrade, last run at 2015-01-28 14:25:30, next run at 2015-01-28 15:25:30 or after
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 319, in on_deactivated
    callback.on_deactivated(v)
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 117, in on_deactivated
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 101, in update_status_bar
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 82, in badge
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 59, in branch
AttributeError: 'NoneType' object has no attribute 'group'
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 303, in on_activated
    callback.on_activated(v)
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 114, in on_activated
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 101, in update_status_bar
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 82, in badge
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 59, in branch
AttributeError: 'NoneType' object has no attribute 'group'
Traceback (most recent call last):
  File "/opt/sublime_text/sublime_plugin.py", line 319, in on_deactivated
    callback.on_deactivated(v)
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 117, in on_deactivated
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 101, in update_status_bar
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 82, in badge
  File "git-statusbar in /home/lbigum/.config/sublime-text-3/Installed Packages/Git-StatusBar.sublime-package", line 59, in branch
AttributeError: 'NoneType' object has no attribute 'group'
```

I modified the following function like so, then it doesn't give a stack trace, but doesn't seem to work either:

```
    def branch(self):
...
                m = re.search(r"\* *\(no branch\) (.*?)", output, flags=re.MULTILINE)

```

I'm using git version 2.1.1 on Ubuntu 14.04.1 LTS

**Comment by randy3k** [comment]
It works with `git version 2.2.2` on my Mac. I will take a look on Ubuntu.

**Comment by randy3k** [comment]
I also found no problems on Ubuntu. Could you give more details on how to reproduce the error?

**Comment by ZenithDK** [comment]
I basically just installed Git-StatusBar via Package Control.
Let me try again now.

**Comment by ZenithDK** [comment]
Okay, so right now it is out-right not working. Any idea on how to debug it?
I've opened the console and I can see that it loads Git-StatusBar, but as I've disabled statusbar_branch and statusbar_status nothing git-related is shown on the statusbar.

**Comment by randy3k** [comment]
Is git binary in the PATH? You could try to specify the path to git in the settings.

**Comment by ZenithDK** [comment]
Git is in my path, yes:
which git
/usr/bin/git

**Comment by randy3k** [comment]
Could you show me the results of `git status` and `git branch` in the directory that sublime is opening.

**Comment by ZenithDK** [comment]
```
lbigum@testpcdk94:~/git(detached*)*$ git status
# HEAD detached at 9f3133c
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#   modified:   something/Android.mk
#
lbigum@testpcdk94:~/git(detached*)*$ git branch
* (detached from 9f3133c)
```

Strange - that looks very different from what I had yesterday where I only made it work after making those changes.
This seems to match what the code is expecting. But it is still not working, though.
![screenshot from 2015-01-29 17 05 53](https://cloud.githubusercontent.com/assets/3715656/5960945/57697896-a7d9-11e4-897b-d991054231f9.png)

**Comment by randy3k** [comment]
You are correct, the results are expected....I have no clue why it is not working though..
